src/jit/lowerxarch.cpp

```diff
@@ -978,6 +978,63 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerHWIntrinsicCC(node, NI_AVX_PTEST, GenCondition::UGT);
             break;
 
+        case NI_FMA_MultiplyAddScalar:
+        {
+            // Math(F).FusedMultiplyAdd is expanded into NI_FMA_MultiplyAddScalar but
+            // depending on additional GT_NEG nodes it can be:
+            //
+            //    x *  y + z -> NI_FMA_MultiplyAddScalar
+            //    x * -y + z -> NI_FMA_MultiplyAddNegatedScalar
+            //   -x *  y + z -> NI_FMA_MultiplyAddNegatedScalar
+            //   -x * -y + z -> NI_FMA_MultiplyAddScalar
+            //    x *  y - z -> NI_FMA_MultiplySubtractScalar
+            //    x * -y - z -> NI_FMA_MultiplySubtractNegatedScalar
+            //   -x *  y - z -> NI_FMA_MultiplySubtractNegatedScalar
+            //   -x * -y - z -> NI_FMA_MultiplySubtractScalar
+            //
+            GenTreeArgList* argList = node->gtGetOp1()->AsArgList();
+            if (argList->Current()->OperIsHWIntrinsic()) // CreateScalarUnsafe is expected
+            {
+                GenTreeUnOp* createScalarOpX = argList->Current()->AsUnOp();
```

Comment:
Nit: I would prefer to use the narrowest possible type here and below:
`GenTreeHWIntrinsic* createScalarOpX = argList->Current()->AsHWIntrinsic();`